detox/src/devices/ios/AppleSimUtils.js

```diff
@@ -248,7 +248,7 @@ class AppleSimUtils {
   }
 
   _joinLaunchArgs(launchArgs) {
-    return _.map(launchArgs, (v, k) => `${k} "${v}"`).join(' ').trim();
+    return _.map(launchArgs, (v, k) => `-${k} "${v}"`).join(' ').trim();
```

Comment:
Since we are making a change here, let's consider the use case where iOS supports a flat list of arguments, as opposed to key-value pairs.

Spreading an array into an object creates the following object:
`{'0':'arg1', '1':'arg2'}`

Perhaps we can catch this inside this map, and instead of producing `-0 arg1 -1 arg2`, just produce `arg1 arg2`.

Reply:
Right now we spread the user provided object (array or object), into the lauchArgs object. This would create a problem if the user wants to provide an array of arguments, rather than a key value object.

Reply:
@LeoNatan this was broken either way (regardless of where the dashes logic is...), are you sure this is the right scope for this?

Reply:
I think that in order to _start_ fixing that we first need to replace the initial args spread with a lodash clone() or smt

Reply:
Not sure if this is the scope. I was just thinking out loud. It depends on how difficult it is to support. If it's a single `if()`, I don't think it's bad to have here. If it requires a full retooling of how we pass arguments top-down, then not the scope.

Reply:
@LeoNatan, I would enhance that in another pull request. Judging from other places in the code, `launchArgs` is supposed to be a key-value map anyways. Making it work as an array will take more effort than changing one line of code.

Reply:
👍

Reply:
BTW I think it would only make sense to allow that if we also support it on Android, otherwise we'd make the API platform specific. This isn't super trivial since on Android we're force onto a key-value scheme. I suppose what makes sense here is to map the array's values onto $key=true pairs...

Reply:
I don’t see a problem with supporting it only on iOS, since Android enforces key/value.